nexus/db-queries/src/db/datastore/project.rs

```diff
@@ -355,23 +356,30 @@ impl DataStore {
         authz_project: &authz::Project,
         pagparams: &PaginatedBy<'_>,
     ) -> ListResultVec<db::model::IpPool> {
-        use db::schema::ip_pool::dsl;
+        use db::schema::ip_pool;
+        use db::schema::ip_pool_resource;
+
         opctx.authorize(authz::Action::ListChildren, authz_project).await?;
+
+        let silo_id = opctx.authn.silo_required().unwrap().id();
+
         match pagparams {
             PaginatedBy::Id(pagparams) => {
-                paginated(dsl::ip_pool, dsl::id, pagparams)
+                paginated(ip_pool::table, ip_pool::id, pagparams)
             }
             PaginatedBy::Name(pagparams) => paginated(
-                dsl::ip_pool,
-                dsl::name,
+                ip_pool::table,
+                ip_pool::name,
                 &pagparams.map_name(|n| Name::ref_cast(n)),
             ),
         }
-        // TODO(2148, 2056): filter only pools accessible by the given
-        // project, once specific projects for pools are implemented
-        // != excludes nulls so we explicitly include them
-        .filter(dsl::silo_id.ne(*INTERNAL_SILO_ID).or(dsl::silo_id.is_null()))
-        .filter(dsl::time_deleted.is_null())
+        .inner_join(ip_pool_resource::table)
+        .filter(
+            ip_pool_resource::resource_type
+                .eq(IpPoolResourceType::Silo)
+                .and(ip_pool_resource::resource_id.eq(silo_id)),
+        )
+        .filter(ip_pool::time_deleted.is_null())
```

Comment:
The logic in this function is correct in the sense that this is giving the list of IP pools available in the current project, but it's silly to write it this way because we are not actually filtering by project. This is pretty old and was meant to be a placeholder for a future in which pools could be linked to projects. I would like to change this to simply be "current silo IP pools". But that is noise and probably belongs in a followup. This PR is, at least, not making anything worse with regard to this function.

Reply:
Ended up doing this in bf2905294649a991acb42f974288c6cac996e2c1 anyway. It was easy and it cut 100 lines net.